src/features/extract_from_files.py

```diff
@@ -1,27 +1,39 @@
 import asyncio
 import base64
+import io
+import logging
 import os
+import typing
 import zipfile
+from concurrent.futures import Executor
+from typing import Optional
 from urllib.parse import urlparse
 
 import PyPDF2
 from aiohttp import ClientSession
 from bs4 import BeautifulSoup
 from pdfminer.high_level import extract_text
-from PyPDF2.utils import PdfReadError
+from PyPDF2.errors import PdfReadError
 
 from app.models import Explanation, StarCase
-from core.extractor import MetadataBase
+from core.extractor import Extractor
 from core.website_manager import WebsiteData
 from lib.settings import RETURN_IMAGES_IN_METADATA
 
-_FOUND_EXTRACTABLE_FILES = "Found extractable files"
-_FOUND_NO_EXTRACTABLE_FILES = "Found no extractable files"
+logger = logging.getLogger(__file__)
 
 
-@MetadataBase.with_key()
-class ExtractFromFiles(MetadataBase):
-    decision_threshold = 0.5
+_CONTENT_SIZE_THRESHOLD = 25  # if a downloaded content is larger than X megabytes it will not be analyzed
```

Comment:
Is there a specific reason for this limitation?

Reply:
Just performance/stability considerations. Imagine there are a couple of 100mb file to process through. 
Even better would be to only download the file completely if it is below that size, but I don't know how.